### ISSUE TITLE
Write uncompressed data for states and nvram

### ISSUE THREAD

**Issue by eadmaster** [finalburnneo/FBNeo#759]
[Followup](https://neo-source.com/index.php?topic=3811.msg31845#msg31845)

This should help sharing the nvram files with MAME and other emulators.

**Comment by barbudreadmon** [comment]
I remember looking further into this, iirc it turned out that even if the data was uncompressed, in many systems there would still be inconsistencies between MAME and FBNeo nvrams, which defeat the purpose of uncompressing them tbh.

**Comment by coyonut** [comment]
I was thinking of taking advantage of what the hi2txt program does with those Mame nvram and if it were possible to do it with FBNeo and its fs or nv. Do you see any possibility? Would those differences make it impossible to use it uncompressed?
It would be great to be able to get the valid data to work as well as it does with .hi (also from FBNeo) or Mame nvram files.
In the end the data of scores inside those nvram should be the same in Mame and FBNeo, shouldn't it be?

**Comment by coyonut** [comment]
yes, it seems logical to maintain backwards compatibilities. maybe as an alternative save format or so that hi2txt can read and understand fs files.
Thanks for answering barbudreadmon ;)

**Comment by barbudreadmon** [comment]
I've an idea to remove headers and uncompress data without breaking backward compatibility, so i'll try to implement it to lessen the discrepancies and ease the parsing of those nvrams, but the compatibility with MAME nvrams **for all emulated systems** won't be guaranteed.

**Comment by coyonut** [comment]
Great barbudreadmon. I hope you get it. Any progress will be welcome, even if it's a handful of more supported games. Thanks!

**Comment by barbudreadmon** [comment]
Libretro port should be fine with https://github.com/libretro/FBNeo/commit/44fabff39fea401297246fe8ad9f0c9aec2658fb

@dinkc64 @tmaul @0xe1f Maybe it should be backported to other ports ? Compiling the `src/burner/nvram.cpp` file i added a while ago is also required.

**Comment by dinkc64** [comment]
Not against the idea, I just don't want to do it until I understand the system better.

**Comment by iq132** [comment]
What about an option to load/save mame nvram (if supported)?

**Comment by barbudreadmon** [comment]
>Not against the idea, I just don't want to do it until I understand the system better.

Nothing fancy : 
- when loading i try the newer unheadered/uncompressed format first (using `BurnNvramLoad`), then fallback to the headered/compressed format (using `BurnStateLoad`)
- when saving i always use the newer unheadered/uncompressed format (using `BurnNvramSave`)

That way, nvrams get automatically converted to the unheadered/uncompressed format.

>What about an option to load/save mame nvram (if supported)?

Not against the idea, the unheadered/uncompressed format is way bigger so it feels like wasting disk space tbh.

**Comment by dinkc64** [comment]
I'm just worried about the bugs that will come with those changes....  I make more bugs when I write code :P

btw there probably is no change in disk usage either way (cluster size)

best regards,
- dink
